ext/drutil/drutil.c

```diff
@@ -71,23 +71,31 @@ static int drutil_init_count;
 #ifdef X86
 
 static inline void
-native_unix_cpuid(uint *eax, uint *ebx, uint *ecx, uint *edx)
+native_unix_cpuid(uint *xax, uint *xbx, uint *xcx, uint *xdx)
 {
 #    ifdef UNIX
-    /* We need to do this ebx trick, because ebx might be used for fPIC,
+    /* We need to do this xbx trick, because xbx might be used for fPIC,
      * and gcc < 5 chokes on it. This can get removed and replaced by
      * a "=b" constraint when moving to gcc-5.
      */
-    asm volatile("xchgl\t%%ebx, %k1\n\t"
+#        ifdef X64
+    asm volatile("xchgq\t%%rbx, %q1\n\t"
+                 "cpuid\n\t"
+                 "xchgq\t%%rbx, %q1\n\t"
+                 : "=a"(*xax), "=&r"(*xbx), "=c"(*xcx), "=d"(*xdx)
+                 : "0"(*xax), "2"(*xcx));
+#        else
+    asm volatile("xchg{l}\t%%ebx, %k1\n\t"
```

Comment:
Have not seen this `{l}` before.  Below doesn't have the brace which is confusing.

Reply:
https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html

Using that same compiler, this code:

"xchg{l}\t{%%}ebx, %1"
corresponds to either

"xchgl\t%%ebx, %1"                 /* att dialect */
"xchg\tebx, %1"                    /* intel dialect */

As discussed offline, I'll remove.